### PR TITLE
formated file name and ids

### PR DESCRIPTION
**Pull request by v-pegao**
1. Reverted the before changes for remove random id in file name as it's causing thread issue and will make build timeout.
2. Renamed `memberUidMapping` to `uidMapping` and make it contains all compounddefId and memberId
3. Add `compounddefIdMapping` which will only contains the compoundefid, it's used to check if renamed name will be duplicated. Seperated this mapping because memberUid will have much duplicated values when ignore case.
4. Remove .xml file which `kind` is `file` or `dir`
5. Add shared extendedIdMappings for file name that extended `({kind})` to safely correct file and uid mapping

### CODE REVIEW

File: src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs [hunk 1]
```diff
@@ -125,6 +107,14 @@ public async Task RunAsync(BuildContext context)
                                   where g.Count() > 1 && duplicate != null
                                   select (string)duplicate.Attribute("refid")).ToList();
 
+            //Get duplicate Ids when ignore case
```
Comment (superyyrrzz) [inline src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs]: >Get [](start = 14, length = 3)

minor: add space after // #Closed

File: src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs [hunk 2]
```diff
@@ -125,6 +107,14 @@ public async Task RunAsync(BuildContext context)
...
+            var results = duplicateItems.Where(id => compounddefIdMapping.ContainsKey(id)).Select(k => compounddefIdMapping.TryRemove(k, out _)).ToList();
+            var duplicatedIds = compounddefIdMapping.GroupBy(k => k.Value.ToLower())
+                             .Where(g => g.Count() > 1)
+                             .Select(kg => kg.Select(kv => kv.Key))
```
Comment (superyyrrzz) [inline src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs]: >kv => kv.Key [](start = 53, length = 12)

Do we need to select kv.Value here? The key has been `ToLower()`, which may be different from the original id. #Closed

File: src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs [hunk 3]
```diff
@@ -125,6 +107,14 @@ public async Task RunAsync(BuildContext context)
...
+                             .SelectMany(ke => ke);
```
Comment (superyyrrzz) [inline src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs]: Add `.ToList()` or `.ToHashSet()` at the end. `SelectMany` returns `IEnumerable<T>`. Each call to `duplicatedIds.Contains` will make this whole logic to run again. #Closed

File: src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs [hunk 4]
```diff
@@ -206,9 +196,25 @@ public async Task RunAsync(BuildContext context)
                         node.Remove();
                     }
                 }
-                doc.Save(Path.Combine(dirInfo.FullName, RegularizeUid(Path.GetFileNameWithoutExtension(p)) + Path.GetExtension(p)));
+
+                string fileName = Path.GetFileNameWithoutExtension(p);
+                if (compounddefIdMapping.TryGetValue(fileName, out string formatedFileName))
+                {
+                    formatedFileName = RegularizeUid(formatedFileName);
+                    if(duplicatedIds.Contains(fileName))
```
Comment (superyyrrzz) [inline src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs]: >( [](start = 22, length = 1)

minor: add space before `(`. #Closed

**Comment by superyyrrzz** [comment]
Add a test case to cover this change? #Closed

File: src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs (hunk 2 above)
Comment (v-pegao) [reply src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs]: `Igroup<string, keyvaluepair<string,string>>.Select(keyvaluepair<string,string> => keyvaluepair<string,string>.Key{get;}) `
it's select the key of the key value pair which is raw key #Closed

File: src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs (hunk 3 above)
Comment (superyyrrzz) [reply src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs]: Got it. You are right.

---
In reply to: [212512868](https://github.com/docascode/code2yaml/pull/17#discussion_r212512868) [](ancestors = 212512868)

File: src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs (hunk 2 above)
Comment (superyyrrzz) [reply src/Microsoft.Content.Build.Code2Yaml.Steps/PreprocessXml.cs]: Got it. You are right.

---
In reply to: [212513538](https://github.com/docascode/code2yaml/pull/17#discussion_r212513538) [](ancestors = 212513538)

**Review by superyyrrzz — APPROVED** [review]

**Pull request merged**